app/lib/scorecard/backend.dart

```diff
@@ -275,12 +275,19 @@ class ScoreCardBackend {
     return PackageStatus.fromModels(p, pv);
   }
 
+  /// Returns whether we should update the [reportType] report for the given
+  /// [packageName] and [packageVersion].
+  ///
+  /// The method will return false, if the package or version does not exists.
+  /// The method will return true, if either of the following is true:
+  /// - it does not have a report yet,
+  /// - the report was updated before [updatedAfter],
+  /// - the report is older than [successThreshold] if it was a success,
+  /// - the report is older than [failureThreshold] if it was a failure.
   Future<bool> shouldUpdateReport(
```

Comment:
Would it be worth it adding tests for some of this logic?

Reply:
I'd rather have a larger integration test on the scheduling and runners with the 4-5 different cases we have. My next planned testing improvement is parsing the source map for browser coverage data, and then I'll take a look on testing these utils too.